testing/trino-product-tests/src/main/java/io/trino/tests/product/snowflake/TestSnowflake.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.snowflake;
+
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.TestGroups.SNOWFLAKE;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+
+public class TestSnowflake
+        extends ProductTest
+{
+    @Test(groups = {SNOWFLAKE, PROFILE_SPECIFIC_TESTS})
+    public void testCreateTableAsSelect()
+    {
+        onTrino().executeQuery("DROP TABLE IF EXISTS snowflake.test.nation");
```

Comment:
Please use `randomTableSuffix()` because Snowflake is shared service.

Reply:
Ok, yes, I'will fix it. 
I followed your Mariadb MR example to add the product test of snowflake. Because snowflake can't be dockerized, I can't add a [EnvMultinodeMariadb.java](https://github.com/trinodb/trino/pull/10046/files#diff-5a6d1f14696842af550834013f3f9e59fe18775e7281fe26ff0ac6dc0a1282bd) for it. Then I compared with bigquery and added just a properties file and updated the [EnvMultinodeAllConnectors.java](https://github.com/trinodb/trino/pull/10046/files#diff-b16ad74ca0f683c7b79e24b243fcd2fa07a2fdf666d7add707ab46ea8f5639f5)

Reply:
Dockerized database isn't mandatory to build environments. You can refer to `EnvMultinodeAzure`.

Reply:
After referring [EnvMultinodeAzure.java ](https://github.com/trinodb/trino/blob/master/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeAzure.java) and the other Envxxx.java in the directory environment, I found that **EnvMultinodeAzure** is also use the hadoopImage and then use a docker container to do the following things. I don't well understand how to implement such an EnvMultinode 😂.. Could you please give me some advices please ?

Reply:
I pushed what I thought was probably the right thing.